cylc/flow/scheduler_cli.py

```diff
@@ -360,6 +376,120 @@ def scheduler_cli(options: 'Values', workflow_id_raw: str) -> None:
     sys.exit(ret)
 
 
+def _resume(workflow_id, options):
+    """Resume the workflow if it is already running."""
+    try:
+        detect_old_contact_file(workflow_id)
+    except ServiceFileError as exc:
+        print(f"Resuming already-running workflow\n\n{exc}")
+        pclient = WorkflowRuntimeClient(
+            workflow_id,
+            timeout=options.comms_timeout,
+        )
+        mutation_kwargs = {
+            'request_string': RESUME_MUTATION,
+            'variables': {
+                'wFlows': [workflow_id]
+            }
+        }
+        pclient('graphql', mutation_kwargs)
+        sys.exit(0)
+
+
+def _version_check(
+    db_file: Path,
+    can_upgrade: bool,
+    can_downgrade: bool
+) -> bool:
+    """Check the workflow can be safely restarted with this version of Cylc."""
+    if not db_file.is_file():
+        # not a restart
+        return True
+    wdbm = WorkflowDatabaseManager(db_file.parent)
+    this_version = parse_version(__version__)
+    last_run_version = wdbm.check_workflow_db_compatibility()
+
+    for itt, (this, that) in enumerate(zip(
```

Comment:
easier to read if `this, that` were `now, before` IMO, or even:

```python
for level, (now, before) in zip(['major', 'minor', 'maint'], zip(x.release, y.release))]:
```

It would make the conditions slightly harder to write, but much easier to read, and more self documenting.

Reply:
IMO "this version" and "that version" make more sense than "now version" and "before version".

Reply:
Happy to leave it to your judgement.